factory/data/dataComponentsHandler_test.go

```diff
@@ -4,46 +4,83 @@ import (
+func TestNewManagedDataComponents(t *testing.T) {
 	t.Parallel()
 
-	coreComponents := componentsMock.GetCoreComponents()
-	shardCoordinator := mock.NewMultiShardsCoordinatorMock(2)
-	args := componentsMock.GetDataArgs(coreComponents, shardCoordinator)
-	args.Config.ShardHdrNonceHashStorage = config.StorageConfig{}
-	dataComponentsFactory, _ := dataComp.NewDataComponentsFactory(args)
-	managedDataComponents, err := dataComp.NewManagedDataComponents(dataComponentsFactory)
-	require.NoError(t, err)
-	err = managedDataComponents.Create()
-	require.Error(t, err)
-	require.Nil(t, managedDataComponents.Blockchain())
+	t.Run("nil factory should error", func(t *testing.T) {
+		t.Parallel()
+
+		managedDataComponents, err := dataComp.NewManagedDataComponents(nil)
+		require.Equal(t, errErd.ErrNilDataComponentsFactory, err)
+		require.Nil(t, managedDataComponents)
+	})
+	t.Run("should work", func(t *testing.T) {
+		t.Parallel()
+
+		coreComponents := componentsMock.GetCoreComponents()
+		shardCoordinator := mock.NewMultiShardsCoordinatorMock(2)
+		args := componentsMock.GetDataArgs(coreComponents, shardCoordinator)
+		dataComponentsFactory, _ := dataComp.NewDataComponentsFactory(args)
+		managedDataComponents, err := dataComp.NewManagedDataComponents(dataComponentsFactory)
+		require.Nil(t, err)
+		require.NotNil(t, managedDataComponents)
+	})
 }
 
-func TestManagedDataComponents_CreateShouldWork(t *testing.T) {
+func TestManagedDataComponents_Create(t *testing.T) {
 	t.Parallel()
 
-	coreComponents := componentsMock.GetCoreComponents()
-	shardCoordinator := mock.NewMultiShardsCoordinatorMock(2)
-	args := componentsMock.GetDataArgs(coreComponents, shardCoordinator)
-	dataComponentsFactory, _ := dataComp.NewDataComponentsFactory(args)
-	managedDataComponents, err := dataComp.NewManagedDataComponents(dataComponentsFactory)
-	require.NoError(t, err)
-	require.Nil(t, managedDataComponents.Blockchain())
-	require.Nil(t, managedDataComponents.StorageService())
-	require.Nil(t, managedDataComponents.Datapool())
+	t.Run("invalid config should error", func(t *testing.T) {
+		t.Parallel()
 
-	err = managedDataComponents.Create()
-	require.NoError(t, err)
-	require.NotNil(t, managedDataComponents.Blockchain())
-	require.NotNil(t, managedDataComponents.StorageService())
-	require.NotNil(t, managedDataComponents.Datapool())
+		coreComponents := componentsMock.GetCoreComponents()
+		shardCoordinator := mock.NewMultiShardsCoordinatorMock(2)
+		args := componentsMock.GetDataArgs(coreComponents, shardCoordinator)
+		args.Config.ShardHdrNonceHashStorage = config.StorageConfig{}
+		dataComponentsFactory, _ := dataComp.NewDataComponentsFactory(args)
+		managedDataComponents, err := dataComp.NewManagedDataComponents(dataComponentsFactory)
+		require.NoError(t, err)
+		err = managedDataComponents.Create()
+		require.Error(t, err)
+		require.Nil(t, managedDataComponents.Blockchain())
+	})
+	t.Run("should work with getters", func(t *testing.T) {
```

Comment:
wrong test name

Reply:
added specific tests for CheckSubcomponents and SetBlockchain